packages/kbn-expandable-flyout/src/index.test.tsx

```diff
@@ -46,14 +55,23 @@ describe('ExpandableFlyout', () => {
   });
 
   it('should render right section', () => {
-    const state = {
-      byId: {
-        [id]: {
-          right: {
-            id: 'key',
+    const state: State = {
+      data: {
+        panels: {
+          byId: {
+            [id]: {
+              right: {
+                id: 'key',
+              },
+              left: undefined,
+              preview: undefined,
+            },
           },
-          left: undefined,
-          preview: undefined,
+        },
+      },
+      ui: {
+        pushVsOverlay: {
+          byId: {},
```

Comment:
there cannot be any default state because we don't know what `urlKey` exists. `urlKey ` for Security Solution would be `flyout` and `timelineflyout` but also could be `memory` for the ephemeris flyouts (like the one used in the rule creation page).

Reply:
hmm..if im understanding @logeekal 's point, the question is should we let each flyout have their own push vs overlay state or should it be shared by all flyouts. i understand this is done because timeline flyout is different. 

Maybe have a check [here](https://github.com/elastic/kibana/pull/192745/files#diff-aefe8a0dff8058c8dda176fc711ed4716dcba65c4b1a99455cfbddbb70c8c55dR67) to bypass the selector value when push vs overlay is disabled? wdyt?
```
const flyoutTypeSetting = useSelector(selectPushVsOverlay());
const flyoutType = flyoutCustomProps?.pushVsOverlay?.disabled ? 'overlay' : flyoutTypeSetting
```
my previous comments about local storage is subject to change then based on this

Reply:
I had thought about this, especially with the potential usage in Discover but had decided to ignore this as I'm hoping the usage in Discover will not happen in the way we had in mind...

I like what you suggested though @christineweng! In my [last commit](https://github.com/elastic/kibana/pull/192745/commits/8e091aa08a04e2ec00357417cee519f3aa5351c0) you'll see that I applied all your suggestions:
- remove `byId` nested inside `pushVsOverlay` slice of state. I'm not simply storing a single value (`push` or `overlay`)
- saving the value in local storage under a single entry `expandableFlyout.ui`. This object currently has a single property (`pushVsOverlay`) and will have in the next PR all the properties for the resize work
- updated all the unit tests accordingly

Reply:
Thank you both 👍